### PR TITLE
Finish ensemble documentation

### LINKED ISSUE #94: Ensemble docs
- Something that needs double-checking and/or conversion to git issues is the dev branch's ensemble.py. ~In particular, I wonder if there's a nice way to do the ensemble.evaluators dict s.t. it hyperlinks to the appropriate functions.~
   - ~On a first sweep, it doesn't look like there's an easy way to include dict members of a specific class with autodoc. Similar approach would be including the functions in a (public) submodule. The docstring could do references to other methods, like, :meth:`ensemble.evaluators.eval_simple_majority`, which would be similar to the current structure. Might want to fool with the names to be user-friendly.~
- The example notebooks (either concept drift and data drift separately, or as a standalone "ensembling" example) should also get some additions.

### PR DESCRIPTION
**Pull request by Anmol-Srivastava**
Closes #94 

### Adds
- example script for ensembles

### Changes
- minor formatting issues for existing docstrings in `ensemble.py`
- modifies how `make_example_data` finds the csv files in the repo. Should work when called in arbitrary directories, now (i.e. correctly).

### CODE REVIEW

**Comment by Anmol-Srivastava** [comment]
Waiting on RTD runs now

File: examples/ensemble_examples.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+print(f"Batch #{0} | Ensemble reference set")
+
+for i, batch in enumerate(df_into_batches[1:]):
+    ensemble.update(batch)
+    print(f"Batch #{i+1} | Ensemble overall drift state: {ensemble.drift_state}")
+
+
+# ## Streaming Ensemble
+# 
+# Using an ensemble of streaming detectors can involve additional features. This example uses both data and concept drift detectors (`KdqTreeStreaming`, `STEPD`), custom subsets of data for different detectors, as well as a different election scheme that can mitigate internal differences in how different detectors 'wait' before alarming for drift.
```
Comment (tms-bananaquit) [inline examples/ensemble_examples.py:79]: The "different election scheme" here sounds like it's referring to ConfirmedApproval. Should the example or the narrative be modified (in the notebook)?

**Review by tms-bananaquit — CHANGES_REQUESTED** [review]
Looks good!

One thing that looks a little wonky is `drift_detector `and `ensemble` being unnavigable because of the way they're set in the class hierarchy.
image

The other consequence of having Ensembles/Elections in a single module is that they're mixed in together within the docs, since members are sorted alphabetically right now. I think it'd feel more natural if they were grouped by Ensemble- and Election-related classes; otherwise, we'd have to maintain the order in the source code (everywhere?) and modify some of the sphinx config.

Maybe it would make sense to have a structure like this? The user has to import Elections from `ensemble` right now anyway, so it doesn't seem like it would be a lot of additional pain.
```
/menelaus/
┖ /drift_detector/
   ┖ menelaus.drift_detector
┖ /ensemble/
   ┖ menelaus.ensemble
   ┖ menelaus.election
```

File: src/menelaus/ensemble.py [hunk 2]
```diff
@@ -319,15 +326,14 @@ def __init__(self, detectors: dict, election, column_selectors: dict = {}):
             detectors (dict): Set of detectors in ensemble. Should be keyed by
```
[the remark this answers is not in the document]
Comment (tms-bananaquit) [reply src/menelaus/ensemble.py:326]: Could also apply to the same kwarg in `StreamingEnsemble`.

File: src/menelaus/ensemble/ensemble.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+                initialized detector object. For instance, ``{'a': ADWIN()}``.
+            election (str): Initialized ``Election`` object for ensemble to evaluate
+                drift among constituent detectors. See implemented election schemes
+                in ``menelaus.ensemble``.
+            columns_selectors (dict, optional): Functions to use
+                for each detector. Functions should take data instance X
+                and return the columns of X that the corresponding detector
+                should operate on. Should match format of ``detectors`` i.e.
+                ``{'a': ADWIN()}`` would need an entry ``{'a': function}`` to
+                use this feature. By default, each no column selection
```
Comment (tms-bananaquit) [inline src/menelaus/ensemble/ensemble.py:86]:
```suggestion
                use this feature. By default, no column selection
```

**Review by tms-bananaquit — APPROVED** [review]
Good job! Two typos to fix and then ready to merge.

**Pull request merged**